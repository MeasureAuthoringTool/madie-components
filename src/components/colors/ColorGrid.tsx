import React from "react";
import tw, { styled } from "twin.macro";
import ColorGridRow, { ColorDefs } from "../../components/colors/ColorGridRow";

export interface ColorGridProps {
  backgroundColor: string;
}

const Wrapper = styled.section<ColorGridProps>`
  background-color: ${(props) => props.backgroundColor};
  ${tw`p-10`}
`;

export default function ColorGrid(props: ColorGridProps) {
  const { backgroundColor } = props;
  return (
    <Wrapper backgroundColor={backgroundColor}>
      <ColorGridRow colorName={ColorDefs.primary} title="Primary" />

      <ColorGridRow colorName={ColorDefs.teal} title="Teal" />

      <ColorGridRow colorName={ColorDefs.green} title="Green" />

      <ColorGridRow colorName={ColorDefs.blue} title="Blue" />

      <ColorGridRow colorName={ColorDefs.yellow} title="Yellow" />

      <ColorGridRow colorName={ColorDefs.red} title="Red" />

      <ColorGridRow colorName={ColorDefs.gray} title="Gray" />
    </Wrapper>
  );
}
