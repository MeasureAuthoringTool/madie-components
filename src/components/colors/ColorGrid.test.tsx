import * as React from "react";
import { render } from "@testing-library/react";
import ColorGrid from "./ColorGrid";

describe("ColorGrid", () => {
  it("should render a ColorGrid, a collection of ColorGridRows", () => {
    const { container } = render(<ColorGrid backgroundColor={"blue"} />);
    expect(container).toBeVisible();
    expect(container.children.length).toBe(1);
    const wrapper = container.children[0];
    expect(wrapper.children.length).toBe(7);
    const colorArray = Array.from(wrapper.children);
    expect(colorArray[0]).toHaveTextContent("Primary");
    expect(colorArray[1]).toHaveTextContent("Teal");
    expect(colorArray[2]).toHaveTextContent("Green");
    expect(colorArray[3]).toHaveTextContent("Blue");
    expect(colorArray[4]).toHaveTextContent("Yellow");
    expect(colorArray[5]).toHaveTextContent("Red");
    expect(colorArray[6]).toHaveTextContent("Gray");
    expect(container).toMatchSnapshot();
  });
});
