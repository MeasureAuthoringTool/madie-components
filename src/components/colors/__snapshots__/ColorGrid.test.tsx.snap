// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorGrid should render a ColorGrid, a collection of ColorGridRows 1`] = `
<div>
  <section
    class="sc-dkPtRN bygvNQ"
  >
    <div
      class="sc-gsDKAQ gflVCz"
    >
      <div
        class="sc-bdvvtL bPdcyv"
      >
        Primary
      </div>
      <div
        class="sc-bdvvtL eKhvDl"
      >
         
      </div>
      <div
        class="sc-bdvvtL cFRocI"
      >
         
      </div>
      <div
        class="sc-bdvvtL hvmxqZ"
      >
         
      </div>
      <div
        class="sc-bdvvtL bTQnKp"
      >
         
      </div>
      <div
        class="sc-bdvvtL leGieO"
      >
         
      </div>
      <div
        class="sc-bdvvtL kPNXfA"
      >
         
      </div>
      <div
        class="sc-bdvvtL ieUjbC"
      >
         
      </div>
      <div
        class="sc-bdvvtL dBEfwz"
      >
         
      </div>
      <div
        class="sc-bdvvtL cagYjH"
      >
         
      </div>
    </div>
    <div
      class="sc-gsDKAQ gflVCz"
    >
      <div
        class="sc-bdvvtL bSWeuc"
      >
        Teal
      </div>
      <div
        class="sc-bdvvtL habBZQ"
      >
         
      </div>
      <div
        class="sc-bdvvtL iIjNP"
      >
         
      </div>
      <div
        class="sc-bdvvtL ijACkf"
      >
         
      </div>
      <div
        class="sc-bdvvtL ypocV"
      >
         
      </div>
      <div
        class="sc-bdvvtL eFQRoT"
      >
         
      </div>
      <div
        class="sc-bdvvtL gkaOFL"
      >
         
      </div>
      <div
        class="sc-bdvvtL elCdnL"
      >
         
      </div>
      <div
        class="sc-bdvvtL hBxkLJ"
      >
         
      </div>
      <div
        class="sc-bdvvtL dyAMRq"
      >
         
      </div>
    </div>
    <div
      class="sc-gsDKAQ gflVCz"
    >
      <div
        class="sc-bdvvtL hmUxPD"
      >
        Green
      </div>
      <div
        class="sc-bdvvtL iPbZNc"
      >
         
      </div>
      <div
        class="sc-bdvvtL kKGxEC"
      >
         
      </div>
      <div
        class="sc-bdvvtL epRcMr"
      >
         
      </div>
      <div
        class="sc-bdvvtL klDEtd"
      >
         
      </div>
      <div
        class="sc-bdvvtL hCuTKD"
      >
         
      </div>
      <div
        class="sc-bdvvtL jBuVG"
      >
         
      </div>
      <div
        class="sc-bdvvtL ghSQuY"
      >
         
      </div>
      <div
        class="sc-bdvvtL grJtOk"
      >
         
      </div>
      <div
        class="sc-bdvvtL enuBaG"
      >
         
      </div>
    </div>
    <div
      class="sc-gsDKAQ gflVCz"
    >
      <div
        class="sc-bdvvtL eobXmq"
      >
        Blue
      </div>
      <div
        class="sc-bdvvtL bqAZgX"
      >
         
      </div>
      <div
        class="sc-bdvvtL ckpPKe"
      >
         
      </div>
      <div
        class="sc-bdvvtL fQeGgs"
      >
         
      </div>
      <div
        class="sc-bdvvtL ioNyfl"
      >
         
      </div>
      <div
        class="sc-bdvvtL corYXh"
      >
         
      </div>
      <div
        class="sc-bdvvtL brgbQE"
      >
         
      </div>
      <div
        class="sc-bdvvtL iALeAY"
      >
         
      </div>
      <div
        class="sc-bdvvtL bDyhHd"
      >
         
      </div>
      <div
        class="sc-bdvvtL VUXhZ"
      >
         
      </div>
    </div>
    <div
      class="sc-gsDKAQ gflVCz"
    >
      <div
        class="sc-bdvvtL itCPfR"
      >
        Yellow
      </div>
      <div
        class="sc-bdvvtL hBWuSK"
      >
         
      </div>
      <div
        class="sc-bdvvtL iufBsZ"
      >
         
      </div>
      <div
        class="sc-bdvvtL cvxnpj"
      >
         
      </div>
      <div
        class="sc-bdvvtL efAKVY"
      >
         
      </div>
      <div
        class="sc-bdvvtL cZmWKK"
      >
         
      </div>
      <div
        class="sc-bdvvtL gzdlPG"
      >
         
      </div>
      <div
        class="sc-bdvvtL jhxjfT"
      >
         
      </div>
      <div
        class="sc-bdvvtL bHccdy"
      >
         
      </div>
      <div
        class="sc-bdvvtL jeXeBb"
      >
         
      </div>
    </div>
    <div
      class="sc-gsDKAQ gflVCz"
    >
      <div
        class="sc-bdvvtL hzpdee"
      >
        Red
      </div>
      <div
        class="sc-bdvvtL fwbqEG"
      >
         
      </div>
      <div
        class="sc-bdvvtL guvoRl"
      >
         
      </div>
      <div
        class="sc-bdvvtL kKVrqx"
      >
         
      </div>
      <div
        class="sc-bdvvtL gygxRj"
      >
         
      </div>
      <div
        class="sc-bdvvtL gBiOlp"
      >
         
      </div>
      <div
        class="sc-bdvvtL jAxUiI"
      >
         
      </div>
      <div
        class="sc-bdvvtL wpzwD"
      >
         
      </div>
      <div
        class="sc-bdvvtL vBDiF"
      >
         
      </div>
      <div
        class="sc-bdvvtL jUbQQV"
      >
         
      </div>
    </div>
    <div
      class="sc-gsDKAQ gflVCz"
    >
      <div
        class="sc-bdvvtL didusF"
      >
        Gray
      </div>
      <div
        class="sc-bdvvtL kOVnlN"
      >
         
      </div>
      <div
        class="sc-bdvvtL kEfvSG"
      >
         
      </div>
      <div
        class="sc-bdvvtL jpUvMe"
      >
         
      </div>
      <div
        class="sc-bdvvtL hbyPgN"
      >
         
      </div>
      <div
        class="sc-bdvvtL eIJqVz"
      >
         
      </div>
      <div
        class="sc-bdvvtL iJOvBd"
      >
         
      </div>
      <div
        class="sc-bdvvtL bNiNtC"
      >
         
      </div>
      <div
        class="sc-bdvvtL hFTQTa"
      >
         
      </div>
      <div
        class="sc-bdvvtL kHoRRT"
      >
         
      </div>
    </div>
  </section>
</div>
`;
